Cesium.CodeGen/Ir/BlockItems/ForStatement.cs

```diff
@@ -28,26 +30,35 @@ public ForStatement(Ast.ForStatement statement)
         IExpression? initExpression,
         IExpression testExpression,
         IExpression? updateExpression,
-        IBlockItem body)
+        IBlockItem body,
+        string breakLabel)
     {
         _initExpression = initExpression;
         _testExpression = testExpression;
         _updateExpression = updateExpression;
         _body = body;
+        _breakLabel = breakLabel;
     }
 
     public IBlockItem Lower(IDeclarationScope scope)
-        => new ForStatement(
-            _initExpression?.Lower(scope),
-            _testExpression.Lower(scope),
-            _updateExpression?.Lower(scope),
-            _body.Lower(scope));
+    {
+        var forScope = new ForScope((IEmitScope)scope);
+        var breakLabel = forScope.GetBreakLabel();
+        scope.AddLabel(breakLabel!);
```

Comment:
I'm not entirely sure of side effects in `Lower`. The idea of `Lower` was that you can call it as much as you like (or even forget it). But now, `item.Lower().Lower()` can break the codegen.

Thoughts?

Reply:
This is unfortunate hack. Proper handling of this requires separate compilation phase where I resolve labels and where I somehow control scope creation. Even if I have everything in place for multi phase compilation now, I do not sure how to properly implement such multiphase processing. I suspect that this clean architecture put performance pressure on us. Other concern over multiple calls to Lower which I have is that in that case we as developers loosing control of what’s going on. 

If you see any tricks, please let me know.

Reply:
Okay, I see that this issue has bigger scope. I'll note some places we'll need to replace in scope of #201.